src/engine/CartesianProductJoin.cpp

```diff
@@ -150,7 +150,7 @@ Result CartesianProductJoin::computeResult(
 
   // Get all child results (possibly with limit, see above).
   for (auto& child : childView()) {
-    if (limitIfPresent.has_value() && child.supportsLimit()) {
```

Comment:
Not sure if this actually fixes a bug (if the code expects the result to be cropped to shape already) or increases processing time by forcing cropping of `IdTable`s that would only be consumed partially anyways.
Thoughts @joka921 ?